crates/nu-protocol/src/config.rs

```diff
@@ -1548,18 +1537,21 @@ fn create_hooks(value: &Value) -> Result<Hooks, ShellError> {
 
             Ok(hooks)
         }
-        v => match v.span() {
-            Ok(span) => Err(ShellError::UnsupportedConfigValue(
-                "record for 'hooks' config".into(),
-                "non-record value".into(),
-                span,
-            )),
-            _ => Err(ShellError::UnsupportedConfigValue(
-                "record for 'hooks' config".into(),
-                "non-record value".into(),
-                Span::unknown(),
-            )),
-        },
+        v => {
+            if let Ok(span) = v.span() {
+                Err(ShellError::UnsupportedConfigValue(
+                    "record for 'hooks' config".into(),
+                    "non-record value".into(),
+                    span,
+                ))
+            } else {
+                Err(ShellError::UnsupportedConfigValue(
+                    "record for 'hooks' config".into(),
+                    "non-record value".into(),
+                    Span::unknown(),
+                ))
+            }
+        }
```

Comment:
I think this is a very complicated way of saying just `Err(...., v.span().unwrap_or_else(Span::unknown))`

Reply:
That is indeed nicer.
Fixed.